documents/write_acls.go

```diff
@@ -0,0 +1,86 @@
+package documents
+
+import (
+	"bytes"
+
+	"github.com/centrifuge/precise-proofs/proofs"
+)
+
+// changedField holds the compact property, old and new value of the field that is changed
+// if the old is nil, then it is a set operation
+// if new is nil, then it is an unset operation
+// if both old and new are set, then it is an edit operation
+type changedField struct {
+	property, old, new []byte
+}
+
+// getChangedFields takes two document trees and returns the compact value of the fields that are changed in new tree.
+// properties may have been added to the new tree or removed from the new tree. Either case, since the new tree is different from old,
+// that is considered a change
```

Comment:
nitpick but maybe this should be on the line above. also, "In either case,".

Reply:
this is not a new line, just a wrap since the comment is long enough.